tests/integration/workload.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+import aiohttp
+
+logger = logging.getLogger(__name__)
+
+
+class Prometheus:
+    """A class that represents a running instance of Prometheus."""
```

Comment:
Could we somehow augment+use the existing `prometheus_server.py`?

Reply:
They are built for different purposes, use different packages , different request processing approaches and will diverge considerably as tests evolve.

Reply:
I'm sort of with @sed-i here - it seems odd to be testing a different `Prometheus` class than the one the charm uses?

Reply:
I think there is something lost in translation here. I am not testing this `Prometheus` class. This `Prometheus` class only provides a growing list of *asynchronous* methods to interact with the deployed Prometheus application. Future PRs will grow this class to query various aspects of Prometheus in ways that are only relevant for integration testing but not for actually managing the Prometheus instance.

Reply:
These methods do not need to be async because nothing else is done in the meanwhile: each and every one of them is awaited (see `asyncio.run`, `asyncio.gather`). Also, each and every one of them is a good fit for `prometheus_server.py`.

Reply:
As discussed in standup this will be explored later.